### PR TITLE
[Native] Add early exit network sample

### PR DESCRIPTION
**Pull request by gichan-jang**
Add early exit ntework sample application with multiple modules.
If the average brightness value of the input video is more than 100,
it does not pass through the next EE module and sends the results immediately.

Signed-off-by: gichan-jang <gichan2.jang@samsung.com>

Self evaluation:
1. Build test: [ *]Passed [ ]Failed [ ]Skipped
2. Run test: [ *]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: native/example_early_exit/example_early_exit.cc [hunk 1]
```diff
@@ -0,0 +1,295 @@
+/**
+ * @file	example_early_exit.c
```
Comment (dongju-chae) [inline native/example_early_exit/example_early_exit.cc:2]: example_early_exit.cc

File: native/example_early_exit/example_early_exit.cc [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+  g_main_loop_run (g_app.loop);
+
+  gst_element_set_state (g_app.pipeline, GST_STATE_NULL);
+
+/* Unregister custom easy filter */
+  NNS_custom_easy_unregister ("CE_passthrough_main");
+  NNS_custom_easy_unregister ("CE_increase_brightness_0");
+  NNS_custom_easy_unregister ("CE_increase_brightness_1");
+  NNS_custom_easy_unregister ("CE_increase_brightness_2");
+  NNS_custom_easy_unregister ("CE_increase_brightness_3");
```
Comment (dongju-chae) [inline native/example_early_exit/example_early_exit.cc:289]: Missing un-registration for `CE_get_average_XXX`?

File: native/example_early_exit/example_early_exit.cc (hunk 1 above)
Comment (gichan-jang) [reply native/example_early_exit/example_early_exit.cc]: file extension is changed to `example_early_exit.c`. Thanks

File: native/example_early_exit/example_early_exit.cc (hunk 2 above)
Comment (gichan-jang) [reply native/example_early_exit/example_early_exit.cc]: Right! Thanks.

**Comment by gichan-jang** [comment]
C-API sample app is uploaded.

File: native/example_early_exit/meson.build [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Install early-exit example
+example_early_exit = executable('nnstreamer_example_early_exit',
+  'example_early_exit.c',
+  dependencies: [glib_dep, gst_dep, gmodule_dep, nns_dep],
+  install: true,
+  install_dir: examples_install_dir
+)
+
+example_early_exit_capi = executable('nnstreamer_example_early_exit_capi',
```
Comment (dongju-chae) [inline native/example_early_exit/meson.build:9]: Well.. I think this needs some if-statement to check `nns_dep` and `nns_capi_dep`. They are declared as `required: false`.

Comment (gichan-jang) [reply native/example_early_exit/meson.build:10]: I added the dependency check. Thanks.

**Review by dongju-chae — APPROVED** [review]
LGTM

**Review by myungjoo — COMMENTED** [review]
Can you use inputCombination/outputCombination of tensor-filter to avoid a tee?
(You will need to draw the diagram carefully so that input/output combination is adequately described.)

**Comment by gichan-jang** [comment]
> Can you use inputCombination/outputCombination of tensor-filter to avoid a tee?
> (You will need to draw the diagram carefully so that input/output combination is adequately described.)

Got it. I will replace `tee` with `outputCombination` from the early exit module.

**Review by myungjoo — APPROVED** [review]

**Pull request merged**